### PR TITLE
Create reusable report component

### PR DESCRIPTION
**Pull request by dhekra-rouatbi**

### CODE REVIEW

File: Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+//
+//  Created by Dhekra Rouatbi on 2/7/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class NextcloudMainViewModel: ReportMainViewModel {
+
```
Comment (Rimktarii) [inline Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift:13]: Maybe renaming this to `NextcloudReportViewModel` ?

File: Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
+//
+//  GDriveMainViewModel.swift
+//  Tella
+//
+//  Created by Dhekra Rouatbi on 2/7/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
```
Comment (Rimktarii) [inline Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift:10]: I think no need for importing Foundation

File: Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
+//
+//  GDriveMainViewModel.swift
+//  Tella
```
Comment (Rimktarii) [inline Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift:3]: Maybe should has same name like the file name

File: Tella/Components/Connections/Card/ConnectionCardDetail.swift [hunk 4]
```diff
@@ -10,18 +10,19 @@ import SwiftUI
...
                 .font(.custom(Styles.Fonts.semiBoldFontName, size: 14))
                 .foregroundColor(.white)
                 .lineLimit(1)
-            
-            Text(subtitle)
-                .font(.custom(Styles.Fonts.regularFontName, size: 12))
-                .foregroundColor(.white)
+            if (subtitle != nil) {
+                Text(subtitle!)
+                    .font(.custom(Styles.Fonts.regularFontName, size: 12))
```
Comment (Rimktarii) [inline Tella/Components/Connections/Card/ConnectionCardDetail.swift:23]: maybe using if let instead and change `subtitle!`

File: Tella/Components/Reports/Draft/DraftView.swift [hunk 5]
```diff
@@ -16,7 +16,8 @@ struct DraftView<VM: DraftViewModelProtocol>: View  {
     
     @EnvironmentObject var mainAppModel: MainAppModel
```
Comment (Rimktarii) [inline Tella/Components/Reports/Draft/DraftView.swift:17]: I think we can remove this  `@EnvironmentObject var mainAppModel: MainAppModel` and use : 
`reportsViewModel.mainAppModel` when it is needed

File: Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift (hunk 2 above)
Comment (dhekra-rouatbi) [reply Tella/Scenes/Nextcloud/ViewModels/NextcloudMainViewModel.swift:10]: I think we need `import Foundation` for `DispatchQueue`

File: Tella/Scenes/CommonConnectionReport/ViewModels/ReportCardViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+         serverName : String?,
+         deleteReport: @escaping (() -> Void)) {
+        
+        let title = report.title ?? ""
+        let serverName = serverName
+        let iconImageName : String? = report.status.iconImageName
+        let listActionSheetItem = report.status.listActionSheetItem
+        let deleteReportStrings = report.status.deleteReportStrings
+        
+        super.init(id: Int(UUID().uuidString) ?? 0,
```
Comment (ValbuenaG) [inline Tella/Scenes/CommonConnectionReport/ViewModels/ReportCardViewModel.swift:26]: Why are we initializing the id like this? Shouldn't we initialize it with the id of the report?

Comment (dhekra-rouatbi) [reply Tella/Scenes/CommonConnectionReport/ViewModels/ReportCardViewModel.swift:26]: yeah, you're right it's a mistake

**Pull request merged**